pager.c

```diff
@@ -2265,27 +2271,35 @@ static void pager_custom_redraw(struct Menu *pager_menu)
  */
 int mutt_pager(const char *banner, const char *fname, PagerFlags flags, struct Pager *extra)
 {
+
+  struct Mailbox *m;
   static char searchbuf[256] = { 0 };
   char buf[1024];
   int ch = 0, rc = -1;
   bool first = true;
   int searchctx = 0;
   bool wrapped = false;
-
   struct Menu *pager_menu = NULL;
+  int msg_in_pager = 0;
   int old_PagerIndexLines; /* some people want to resize it while inside the pager */
 #ifdef USE_NNTP
   char *followup_to = NULL;
 #endif
 
-  struct Mailbox *m = ctx_mailbox(Contex2);
-
   if (!(flags & MUTT_SHOWCOLOR))
     flags |= MUTT_SHOWFLAT;
 
   int index_space = C_PagerIndexLines;
-  if (extra->ctx && extra->ctx->mailbox)
+ 
+  // Do we even consire an option when pager is invoked without mailbox?
+  assert(extra);
+  assert(extra->ctx);
+  assert(extra->ctx->mailbox);
+  if (extra->ctx && extra->ctx->mailbox) 
```

Comment:
@flatcap can what do you think about the above?

Reply:
this is apparently not a good idea. This breaks helper since mailbox is NULL in that case. I will remove the last two asserts